Src/CSharpier/SyntaxPrinter/Modifiers.cs

```diff
@@ -2,34 +2,92 @@ namespace CSharpier.SyntaxPrinter;
 
 internal static class Modifiers
 {
-    public static Doc Print(SyntaxTokenList modifiers, FormattingContext context)
+    private class DefaultOrder : IComparer<string>
     {
-        if (modifiers.Count == 0)
+        // use the default order from https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0036
+        private static readonly string[] DefaultOrdered =
         {
-            return Doc.Null;
+            "public",
+            "private",
+            "protected",
+            "internal",
+            "file",
+            "static",
+            "extern",
+            "new",
+            "virtual",
+            "abstract",
+            "sealed",
+            "override",
+            "readonly",
+            "unsafe",
+            "required",
+            "volatile",
+            "async",
+        };
+
+        public int Compare(string? x, string? y)
+        {
+            int GetIndex(string? value)
+            {
+                var result = Array.IndexOf(DefaultOrdered, value);
+                return result == -1 ? int.MaxValue : result;
+            }
+
+            return GetIndex(x) - GetIndex(y);
         }
+    }
+
+    private static readonly DefaultOrder Comparer = new();
 
-        return Doc.Group(Doc.Join(" ", modifiers.Select(o => Token.Print(o, context))), " ");
+    public static Doc Print(SyntaxTokenList modifiers, FormattingContext context)
+    {
+        return PrintWithSortedModifiers(
+            modifiers,
+            sortedModifiers =>
+                Doc.Group(Doc.Join(" ", sortedModifiers.Select(o => Token.Print(o, context))), " ")
+        );
     }
 
     public static Doc PrintWithoutLeadingTrivia(
         SyntaxTokenList modifiers,
         FormattingContext context
     )
+    {
+        return PrintWithSortedModifiers(
+            modifiers,
+            sortedModifiers =>
+                Doc.Group(
+                    Token.PrintWithoutLeadingTrivia(sortedModifiers.First(), context),
+                    " ",
+                    sortedModifiers.Count() > 1
```

Comment:
Doesn't this give a "multiple enumeration of IEnumerable" error?
It might be better to make `PrintWithSortedModifers()` call `ToArray()` on `sortedModifers` and pass it as an `IReadOnlyList` instead of an `IEnumerable`.